docs: add mm structured outputs tutorial

everettVT opened the pull request:
## Changes Made

add the markdown notebook tutorial to the multimodal structured outputs blog post and pipeline.

ykdojo commented on docs/examples/mm_structured_outputs.md:
````diff
@@ -0,0 +1,431 @@
+
+```python
+df_raw = daft.read_huggingface("HuggingFaceM4/the_cauldron/ai2d").limit(LIMIT).collect()
+df_raw.show(3)
````
show what the output is supposed to look like here.

ykdojo commented on docs/examples/mm_structured_outputs.md:
```diff
@@ -0,0 +1,431 @@
+## 3. Preprocessing
+
+We need to:
+1. Decode images into Daft's Image type
```
This is not showing up correctly on the rendered HTML

ykdojo commented on docs/examples/mm_structured_outputs.md:
```diff
@@ -0,0 +1,431 @@
+| Structured Outputs (with image) | `run_inference(with_image=True)` | Predict multiple-choice letter with the image attached |
+| Ablation (no image) | `run_inference(with_image=False)` | Predict the same question *without* the image |
+| Quadrant classification | `classify_quadrants()` | Bucket behavior into Both Correct / Image Helped / Image Hurt / Both Incorrect |
+| LLM-as-a-Judge | `run_judge()` | Diagnose failure modes on the “Image Hurt” + “Both Incorrect” subsets |
```
LLM as a judge or a VLM as a judge? It'll be helpful to be consistent.

ykdojo commented on docs/examples/mm_structured_outputs.md:
````diff
@@ -0,0 +1,431 @@
+```python
+subsets = ["ai2d", "chartqa", "docvqa", "infographicvqa", ...]
+for subset in subsets:
+    df = run_full_pipeline(subset, MODEL_ID)
````
I think it'll be helpful to explain what this function is supposed to do here because this is the first time you mentioned this particular function.

Plus this is still in a copy and pasteable code block even though it's pseudocode.

everettVT commented:
Thanks for the review. All feedback addressed and tutorial cleaned up for clarity:

**Fixed per review:**
- Replaced "notebook" terminology with "tutorial" throughout
- Added `.env` file setup instructions with HuggingFace token guidance
- Improved dataset output description with specific column structure
- Fixed numbered list rendering in preprocessing section
- Made LLM/VLM-as-a-Judge terminology consistent (VLM throughout)
- Removed redundant comments

**Additional improvements:**
- Added key insight from blog post: ~70% of correct answers don't actually require images, motivating why ablation study matters
- Replaced confusing production pipeline mapping table with simpler About section
- Consolidated repetitive content in Scale section
- Removed comments that just restate what code obviously does
- Merged related code blocks to reduce fragmentation

All changes verified against the [blog post](https://www.daft.ai/blog/multimodal-structured-outputs-evaluating-vlm-image-understanding-at-scale) and [production pipeline](https://github.com/Eventual-Inc/daft-examples/blob/main/use_cases/image_understanding_eval/eval_image_understanding.py) for technical accuracy.

Verified with `make docs` and `make doctests`.

ykdojo commented on docs/examples/mm_structured_outputs.md:
```diff
@@ -0,0 +1,387 @@
+# Evaluating Image Understanding at Scale with Structured Outputs
+
+*An end-to-end example of **Multimodal Structured Outputs** with Daft and Qwen3-VL-8B*
```
lowercase Multimodal Structured Outputs

ykdojo commented on docs/examples/mm_structured_outputs.md:
```diff
@@ -0,0 +1,387 @@
+Our pipeline will:
+
+1. Run structured output inference on image+text prompts
+2. Conduct an **ablation study** (with vs. without images) to isolate image understanding
```
Would be helpful to explain this a little bit more by saying something like "In other words, does it actually understand the images or does it cheat by looking at the text?"

Maybe as a sub bullet, I'm not sure.
